PhpcsChanged/Cli.php

```diff
@@ -233,29 +236,42 @@ function runSvnWorkflowForFile(string $svnFile, array $options, ShellOperator $s
 		if (! $shell->isReadable($svnFile)) {
 			throw new ShellException("Cannot read file '{$svnFile}'");
 		}
-		$svnFileInfo = getSvnFileInfo($svnFile, $svn, [$shell, 'executeCommand'], $debug);
-		$unifiedDiff = getSvnUnifiedDiff($svnFile, $svn, [$shell, 'executeCommand'], $debug);
-		$revisionId = getSvnRevisionId($svnFileInfo);
-		$isNewFile = isNewSvnFile($svnFileInfo);
 
-		$newFileHash = $shell->getFileHash($svnFile);
-		$newFilePhpcsOutput = isCachingEnabled($options) ? $cache->getCacheForFile($svnFile, 'new', $newFileHash, $phpcsStandard ?? '') : null;
+		$newFileHash = '';
+		$newFilePhpcsOutput = null;
+		if (isCachingEnabled($options)) {
+			$newFileHash = $shell->getFileHash($svnFile);
+			$newFilePhpcsOutput = $cache->getCacheForFile($svnFile, 'new', $newFileHash, $phpcsStandard ?? '');
+		}
 		if ($newFilePhpcsOutput) {
-			$debug("Using cache for new file '{$svnFile}' at revision '{$revisionId}', hash '{$newFileHash}', and standard '{$phpcsStandard}'");
+			$debug("Using cache for new file '{$svnFile}', hash '{$newFileHash}', and standard '{$phpcsStandard}'");
 		}
 		if (! $newFilePhpcsOutput) {
-			$debug("Not using cache for new file '{$svnFile}' at revision '{$revisionId}', hash '{$newFileHash}', and standard '{$phpcsStandard}'");
+			$debug("Not using cache for new file '{$svnFile}', hash '{$newFileHash}', and standard '{$phpcsStandard}'");
 			$newFilePhpcsOutput = getSvnNewPhpcsOutput($svnFile, $phpcs, $cat, $phpcsStandardOption, [$shell, 'executeCommand'], $debug);
 			if (isCachingEnabled($options)) {
 				$cache->setCacheForFile($svnFile, 'new', $newFileHash, $phpcsStandard ?? '', $newFilePhpcsOutput);
 			}
 		}
-		$fileName = DiffLineMap::getFileNameFromDiff($unifiedDiff);
+
+		$fileName = $shell->getFileNameFromPath($svnFile);
 		$newFilePhpcsMessages = PhpcsMessages::fromPhpcsJson($newFilePhpcsOutput, $fileName);
-		$hasNewFilePhpcsIssues = !empty($newFilePhpcsMessages->getMessages());
+		$hasNewPhpcsMessages = !empty($newFilePhpcsMessages->getMessages());
+
+		if (! $hasNewPhpcsMessages) {
+			throw new NoChangesException("New file '{$svnFile}' has no PHPCS messages; skipping");
```

Comment:
It's not an issue of this PR, but we seem to be using "new file" in multiple meanings across the codebase, which might be confusing when reading the output.

Perhaps we should be using "new version of the {$svnFile}" here, similarly to line 272 where "old file version" is used. And limit the usage of "new file" to newly added file? Or perhaps even better, use "newly added file" instead of "new file"?

Reply:
Yeah, I agree! I realized this same issue when working on this PR and I wanted to go through and rename everything, but I decided it would be better for a different PR. I'll get that next.